File: aes_base64.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
'''
### How to run:
pip install pycrypto
python3 aes_base64.py
'''


from Crypto.Cipher import AES
import base64


class AESEncrypto:
    '''
    AES/CBC/PKCS5Padding
    '''

    def __init__(self, key, iv):
        self.BLOCK_SIZE = 16
        self.raw_key = base64.b64decode(key)
        self.iv = iv

    def pkcs5_pad(self, s):
        return s + (self.BLOCK_SIZE - len(s) % self.BLOCK_SIZE) * \
            chr(self.BLOCK_SIZE - len(s) % self.BLOCK_SIZE)

    def pkcs5_unpad(self, s):
        return s[0:-ord(s[-1])]

    def encrypt(self, value):
        '''
        input str, return base64 string
        '''
        cipher = AES.new(self.raw_key, AES.MODE_CBC, self.iv)
        value = self.pkcs5_pad(value)
        crypted = base64.b64encode(cipher.encrypt(value)).decode()
        return crypted

    def decrypt(self, value):
        '''
        input base64 string, return bytes
        '''
        cipher = AES.new(self.raw_key, AES.MODE_CBC, self.iv)
        _value = base64.b64decode(value)
        crypted = cipher.decrypt(_value)
        if type(crypted) is bytes:
            crypted = crypted.decode('raw_unicode_escape')
        return self.pkcs5_unpad(crypted)


if __name__ == '__main__':
    # init values
    content_str = "this is a test string"
    aes_key = "cTFXd1RySmQ5VzZHa0NoeQ=="
    iv_str = "1234567890864210"
    print("Origin: %s" % content_str)

    # init obj
    aes = AESEncrypto(aes_key, iv_str)
    # encryption
    _en_str = aes.encrypt(content_str)
    # decryption
    print("Encrypt: %s" % _en_str)
    print("Decrypt: %s" % aes.decrypt(_en_str))

File: AESBase64.java
/**
 * Created by zeeler on 2019-10-1.
 * How to run:
 * 0) Make sure you have jre or jdk installed on your operation system
 * java -version
 * 1) compile code
 * javac -cp .:org-apache-commons-codec.jar AESBase64.java
 * 2) run class
 * java -cp .:org-apache-commons-codec.jar AESBase64
 */

import org.apache.commons.codec.binary.Base64;
import java.io.UnsupportedEncodingException;
import javax.crypto.Cipher;
import javax.crypto.IllegalBlockSizeException;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;

public class AESBase64{
    // algorithm and padding mode
    public static final String ALGORITHM_PADDING = "AES/CBC/PKCS5Padding";

    private static final String coding = "UTF-8";

    private SecretKeySpec keySpec;
    private IvParameterSpec iv; 

    public AESBase64(String aesKey, String ivStr) throws SecurityException, UnsupportedEncodingException {
        if (aesKey == null) {
            throw new SecurityException("AES key is null");
        }
        if (ivStr == null) {
            throw new SecurityException("IV is null");
        }
        iv = new IvParameterSpec(ivStr.getBytes(coding));
        byte[] raw = Base64.decodeBase64(aesKey.getBytes(coding));
        if (raw == null || raw.length != 16) {
            throw new IllegalArgumentException("AES Key is invalid");
        }

        keySpec = new SecretKeySpec(raw, "AES");
    }

    // decryption
    public String decrypt(String base64EncodeData) throws SecurityException {
        try {

            byte[] encryptedByte = Base64.decodeBase64(base64EncodeData.getBytes(coding));
            byte[] decryptedByte = decrypt(encryptedByte);
            return new String(decryptedByte, coding);
        } catch (UnsupportedEncodingException e) {
            throw new SecurityException(e);
        }
    }

    public byte[] decrypt(byte[] cipherData) throws SecurityException {
        Cipher cipher = null;
        try {
            cipher = Cipher.getInstance(ALGORITHM_PADDING);
            cipher.init(Cipher.DECRYPT_MODE, keySpec, iv);
            if (null == cipherData) {
                throw new IllegalBlockSizeException("no block data");
            }
            return cipher.doFinal(cipherData);
        } catch (Exception e) {
            throw new SecurityException(e);
        }
    }

    // encryption
    public String encrypt(String data) throws SecurityException {
        try {
            byte[] rawData = data.getBytes(coding);
            return new String(Base64.encodeBase64(encrypt(rawData)), coding);
        } catch (UnsupportedEncodingException e) {
            throw new SecurityException(e);
        }
    }

    public byte[] encrypt(byte[] rawData) throws SecurityException {
        Cipher cipher = null;
        try {
            cipher = Cipher.getInstance(ALGORITHM_PADDING);
            cipher.init(Cipher.ENCRYPT_MODE, keySpec, iv);
            return cipher.doFinal(rawData);
        } catch (Exception e) {
            throw new SecurityException(e);
        }
    }

    // main function
    public static void main(String args[]){
        try {
            // init values
            String contentStr = "this is a test string";
            String aesKey = "cTFXd1RySmQ5VzZHa0NoeQ==";
            String ivStr = "1234567890864210";
            System.out.println("Origin: " + contentStr);

            // init obj
            AESBase64 aesCoder = new AESBase64(aesKey, ivStr);
            // encyption
            String encStr = aesCoder.encrypt(contentStr);
            // decryption
            String decStr = aesCoder.decrypt(aesCoder.encrypt(contentStr));
            System.out.println("Encrypt: " + encStr);
            System.out.println("Decrypt: " + decStr);

        } catch (Exception e) {
            System.out.println(e);
        }
    }

}
